### PR TITLE
PR - Add project feature

### PR DESCRIPTION
**Pull request by Tom-Sarfo**
- The Add Project feature allows users to create a project by adding team members and providing a name for the project
- This will allow users to keep track of tasks for a unique project.

- Made use of React router dom for page navigation
- Made use of  React Context API and the Reducer hook for state management

### CODE REVIEW

File: src/ProjectLayout/CreateProjectForm.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { useNavigate } from 'react-router-dom';
+import { 
+   ProjectDispatch 
+  } from '../stateManagement/ProjectContext';
+
+export default function CreateProjectForm() {
+  const dispatch = useContext(ProjectDispatch)
+  const [projectName, setProjectName] = useState('');
+  const [ members, setMembers ] = useState([])
+  const navigate = useNavigate()
```
Comment (noramens) [inline src/ProjectLayout/CreateProjectForm.js]: @Tom-Sarfo its more elegant to separate your other hooks from your states

File: src/ProjectLayout/MemberForm.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        name: member,
+      },
+        ...members
+      ]);
+      e.target.reset();
+    }
+    
+  return (
+    <div>
+      <form onSubmit={e => handleSubmit(e)}>
```
Comment (noramens) [inline src/ProjectLayout/MemberForm.js]: @Tom-Sarfo writing your code this way would have still sufficed:

`<form onSubmit={handleSubmit}>`

File: src/ProjectLayout/MemberForm.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        type='text' 
+        name='memberName' 
+        id='memberName' 
+        placeholder='Add team Member(s)'
+        onChange={e => setMember(e.target.value)}
+        />
+        <button type='submit' id='add-member-btn'>Add</button>
+      </form>
+      <div className='chips-container'>
+        { members.map(myMember => (
```
Comment (noramens) [inline src/ProjectLayout/MemberForm.js]: @Tom-Sarfo you should check for the availability of a variable before mapping on it. You can use optional chaining in this scenario.

File: src/ProjectLayout/ProjectBoard.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+          <Link to={`/projects/${project.id}`}  key={project.id}><ProjectCard project = {project} /></Link>
+        ))}
+      </div>
+        
+        <Link to="/board" className="float-btn">
+         <div className="float-btn-container"><span>Add Project</span></div>
+        </Link>
+    </main>
+  )
+}
```
Comment (noramens) [inline src/ProjectLayout/ProjectBoard.js:29]: @Tom-Sarfo consider using `Prettier` extension on vscode to help automatically format your code when you save it

Comment (Tom-Sarfo) [reply src/ProjectLayout/ProjectBoard.js:29]: Okay
I have it installed though. I will configure it to format onsave

File: src/ProjectLayout/MemberForm.js (hunk 3 above)
Comment (Tom-Sarfo) [reply src/ProjectLayout/MemberForm.js]: Okay sure.

File: src/ProjectLayout/CreateProjectForm.js (hunk 1 above)
Comment (Tom-Sarfo) [reply src/ProjectLayout/CreateProjectForm.js]: Sure

**Pull request merged**